include/lsst/afw/table/FieldBase.h

```diff
@@ -211,7 +211,7 @@ struct FieldBase< Array<U> > {
 
     template <typename Derived>
     void setValueDeep(Element * p, ndarray::ExpressionBase<Derived> const & value) const {
-        if (value.template getSize<0>() != _size) {
+        if (value.template getSize<0>() != static_cast<std::size_t>(_size)) {
```

Comment:
Your instincts weren't at all wrong; it was actually a historical decision more than a deliberate one, and I liked your proposal to switch enough that I started looking into doing it here before I realized there was a closely-related type I couldn't easily change.  But it is more of a change than I'd like to do now.  I'm planning to do a bit of refactoring of afw::table in the not-too-distance future, and I'll include another look at all of these types in that.